dask_expr/_shuffle.py

```diff
@@ -615,12 +619,23 @@ class AssignPartitioningIndex(Blockwise):
         Number of partitions after repartitioning is finished.
     """
 
-    _parameters = ["frame", "partitioning_index", "index_name", "npartitions_out"]
+    _parameters = [
+        "frame",
+        "partitioning_index",
+        "index_name",
+        "npartitions_out",
+        "assign_index",
+    ]
 
     @staticmethod
-    def operation(df, index, name: str, npartitions: int):
+    def operation(df, index, name: str, npartitions: int, assign_index):
         """Construct a hash-based partitioning index"""
-        index = _select_columns_or_index(df, index)
+        if assign_index:
+            index = df[[]]
+            index["_index"] = df.index
```

Comment:
Might be useful to drop a comment here to clarify that we need to convert the index to a frame because `partitioning_index` will ignore the index? (Only saying this because I was initially confused by this code block.

Note that we could probably do something like `index = df.index.to_frame()`, but I don't think it really matters.

Reply:
Yeah comment is a good idea


```
index = df.index.to_frame()
```

That doesn't work (was my first idea as well), but to_frame loses the index which will cause alignment issues later on...

Reply:
Okay cool, thanks for checking